Clarify params and query merging (#7744)
zeit/next.js: packages/next/README.md

The matched path parameter will be sent as a query parameter to the page.

For example, the route `/post/1` will have the following `query` object: `{ pid: '1' }`.

> Similarly, the route `/post/abc?foo=bar` will have the `query` object: `{ foo: 'bar', pid: 'abc' }`. However, if your query and route params name are the same. Route parameters are going to override query params with the same name. For example `/post/abc?pid=bcd` will have the `query` object: `{ pid: 'abc' }`.

Reviewer: ```suggestion
Similarly, the route `/post/abc?foo=bar` will have the `query` object: `{ foo: 'bar', pid: 'abc' }`. However, if a query and route param name are the same, route parameters will override the matching query params. For example `/post/abc?pid=bcd` will have the `query` object: `{ pid: 'abc' }`.
```
